src/core/connection.c

```diff
@@ -2520,26 +2520,16 @@ QuicConnProcessPeerVersionNegotiationTP(
                 break;
             }
         }
-        if (CurrentVersionIndex == SupportedVersionsLength) {
-            //
-            // Current version not supported, start incompatible version negotiation.
-            //
-            CXPLAT_DBG_ASSERTMSG(FALSE, "Should be caught in binding layer");
-            return QUIC_STATUS_VER_NEG_ERROR;
-        }
+        CXPLAT_FRE_ASSERTMSG(CurrentVersionIndex != SupportedVersionsLength, "Incompatible Version Negotation should happen in binding layer");
```

Comment:
Does this need to be a FRE assert? In other words, are you confident to say this won't happen, and if it does, Windows should bugcheck?